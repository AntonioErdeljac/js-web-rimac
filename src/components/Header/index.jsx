import React, { memo } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import Message from '../Message';

import { useSystem } from '../../providers';

import './index.scss';

const Header = () => {
  const system = useSystem();

  return (
    <div className="r-header d-inline-flex align-items-center justify-content-center justify-content-lg-between px-5">
      <div className="r-header-menu">
        <FontAwesomeIcon size="lg" icon="bars" />
      </div>
      <h4 className="text-primary text-monospace">CONCEPT_TWO</h4>
      <Message
        variant={system.getStatus()}
        icon="exclamation-circle"
        title={system.getStatus()}
        description={system.getAlertMessage()}
      />
    </div>
  );
};

export default memo(Header);
