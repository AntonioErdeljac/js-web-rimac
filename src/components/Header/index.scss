.r-header {
  height: 88px;
  width: 100%;
  background-color: #DEE0F0;
}
