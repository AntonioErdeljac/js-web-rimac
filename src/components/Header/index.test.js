import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import Header from './index';

const mockGetStatus = jest.fn();
const mockGetAlertMessage = jest.fn();

jest.mock('../../providers', () => ({
  useSystem: () => ({
    getStatus: mockGetStatus,
    getAlertMessage: mockGetAlertMessage,
  }),
  useDrawer: () => ({
    toggle: () => {},
    isOpen: false,
  }),
}));

const renderComponent = () => shallow(<Header />);

describe('Header Component', () => {
  it('renders correctly according to snapshot', () => {
    expect(toJson(renderComponent())).toMatchSnapshot();
  });

  it('calls getStatus and getAlertMessage on mount', () => {
    renderComponent();

    expect(mockGetStatus).toHaveBeenCalled();
    expect(mockGetAlertMessage).toHaveBeenCalled();
  });
});
