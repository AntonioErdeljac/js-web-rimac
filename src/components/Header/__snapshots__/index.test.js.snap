// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly according to snapshot 1`] = `
<div
  className="r-header d-inline-flex align-items-center justify-content-center justify-content-lg-between px-5"
>
  <div
    className="r-header-menu"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="bars"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="lg"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </div>
  <h4
    className="text-primary text-monospace"
  >
    CONCEPT_TWO
  </h4>
  <Memo(Message)
    icon="exclamation-circle"
  />
</div>
`;
